src/API.jl

```diff
@@ -1016,6 +1016,13 @@ function precompile(ctx::Context; internal_call::Bool=false, kwargs...)
         push!(direct_deps, Base.PkgId(ctx.env.pkg.uuid, ctx.env.pkg.name))
     end
 
+    if debugmode
+        str = sprint() do io
+            show(io, MIME"text/plain"(), depsmap)
+        end
+        @debug "Full depsmap:\n$str"
+    end
+
```

Comment:
```suggestion
    @debug "Precompilation dependency map" depsmap=sprint(show, MIME"text/plain"(), depsmap)
```

Reply:
I didn't realize the `@debug` guarded against the `sprint(show, MIME"text/plain"(), depsmap)` being evaluated if not in debug mode. This is tidier, thanks